sdk/nodejs/resource.ts

```diff
@@ -99,6 +99,14 @@ export abstract class Resource {
                 const provider = (<CustomResourceOptions>opts).provider;
                 if (provider === undefined) {
                     (<CustomResourceOptions>opts).provider = opts.parent.getProvider(t);
+                } else {
+                    // If a provider was specified, add it to the providers map under this type's package so that
+                    // any children of this resource inherit its provider.
+                    const typeComponents = t.split(":");
+                    if (typeComponents.length === 3) {
+                        const pkg = typeComponents[0];
+                        this.__providers = { ...this.__providers, pkg: provider };
```

Comment:
this should be; `this.__providers = { ...this.__providers, [pkg]: provider };`  otherwise you're literally just makinga  property called 'pkg' instead of a property with the actual name of the package.

Reply:
Hah, good catch